concent_api/core/models.py

```diff
@@ -485,3 +493,478 @@ def clean(self):
+    deadline = DateTimeField(blank = True, null = True)
+    src_code = TextField()
+    extra_data = JSONField()
+    short_description = TextField()
+    working_directory = TextField()
+    performance = DecimalField(decimal_places=2, max_digits=3)
+    docker_images = JSONField()
+
+    if not len(ITEMS) == len(message.tasks.ComputeTaskDef().ITEMS.keys()):
+        raise ValidationError({
+            'ComputeTaskDef': 'StoredComputeTaskDef model has not same fields as golem_messages.message.tasks.ComputeTaskDef()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.ComputeTaskDef().ITEMS.keys()):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'ComputeTaskDef': 'StoredComputeTaskDef model has not same fields as golem_messages.message.tasks.ComputeTaskDef()'
+            })
+    '''
+    VALIDATION FOR task_id
+    VALIDATION FOR subtask_id
+    VALIDATION FOR deadline
+    VALIDATION FOR src_code
+    VALIDATION FOR extra_data
+    VALIDATION FOR short_description
+    VALIDATION FOR working_directory
+    VALIDATION FOR performance
+    VALIDATION FOR docker_images
+    '''
+
+
+class StoredTaskToCompute(AbstractMessageModel):
+
+    ITEMS = [
+        'requestor_id',
+        'requestor_public_key',
+        'requestor_ethereum_public_key',
+        'provider_id',
+        'provider_public_key',
+        'provider_ethereum_public_key',
+        'compute_task_def',
+        'package_hash',
+        'concent_enabled',
+        'price',
+        'header',
+        'sig',
+    ]
+
+    requestor_id = BinaryField()
+    requestor_public_key = BinaryField()
+    requestor_ethereum_public_key = BinaryField()
+    provider_id = BinaryField()
+    provider_public_key = BinaryField()
+    provider_ethereum_public_key = BinaryField()
+    compute_task_def = ForeignKey(StoredComputeTaskDef)
+    package_hash = CharField(max_length=HASH_LENGTH)
+    concent_enabled = BooleanField(default=False)
+    price = PositiveIntegerField()
+
+    if not len(ITEMS) == len(message.tasks.TaskToCompute().__slots__):
+        raise ValidationError({
+            'StoredTaskToCompute': 'StoredTaskToCompute model has not same fields as golem_messages.message.tasks.TaskToCompute()'
+        })
+
+    for model_item, golem_message_item in zip(ITEMS, message.tasks.TaskToCompute().__slots__):
+        if not model_item == golem_message_item:
+            raise ValidationError({
+                'StoredTaskToCompute': 'StoredTaskToCompute model has not same fields as golem_messages.message.tasks.TaskToCompute()'
+            })
+
+    def clean(self):
+        if not len(self.requestor_id) == GOLEM_PUBLIC_KEY_LENGTH or not isinstance(self.requestor_id, bytes):
+            raise ValidationError({
+                'requestor_id': f'requestor_id must be bytes instance and must have length equal to {GOLEM_PUBLIC_KEY_LENGTH}'
+            })
```

Comment:
Since it's `BinaryField`, the `not isinstance(self.requestor_id, bytes)` can't be false - it should be an `assert`.

Same in other validations below.

Also, this is a bit repetitive. You could check the lengths of all those fields in a loop.

Reply:
Yes, lets make it similar to what we wanted to do with system checks, reusable atomic generic validators.